src/main/java/nova/core/block/Block.java

```diff
@@ -52,6 +53,10 @@ public final BlockTransform transform() {
 		return get(BlockTransform.class);
 	}
 
+	public final BlockMaterial material() {
+		return getOrAdd(new BlockMaterial());
```

Comment:
Blocks need a material in the super constructor(in MC at least), so it's either this or an null check at the other end.